postgres-derive/src/fromsql.rs

```diff
@@ -233,16 +247,15 @@ fn composite_body(ident: &Ident, fields: &[Field]) -> TokenStream {
     }
 }
 
-fn build_generics(source: &Generics) -> (Generics, Lifetime) {
-    // don't worry about lifetime name collisions, it doesn't make sense to derive FromSql on a struct with a lifetime
-    let lifetime = Lifetime::new("'a", Span::call_site());
-
+fn build_generics(source: &Generics, borrowed_lifetimes: HashSet<Lifetime>) -> (Generics, Lifetime) {
+    // This is the same parent lifetime name serde uses
+    let lifetime = Lifetime::new("'de", Span::call_site());
+    // Sort lifetimes for deterministic code-gen
```

Comment:
This could also preserve the declared order of lifetimes rather than going through a hashset.

Reply:
You're right. I was over-optimizing on the extraction path to handle lifetime dedup. In the real-world though, a simple `borrowed_lifetimes.contains(lifetime)` shouldn't ever be a problem. I made the changes to go through a vec instead of a hashset and preserve declaration order.